### PR TITLE
hashing

### PR DESCRIPTION
**Pull request by omarelsheekh**

### CODE REVIEW

**Comment by abom** [comment]
All hash methods return a hex digest, what if i need to feed the data like you did [here](https://github.com/js-next/js-ng/pull/10/files#diff-a06c08da8a479bf9cb2af44a11717450R241)?

I need to be able to get a specific algorithm, update it with data, and get the hex digest at the end.

**Comment by abom** [comment]
Also, encoding should not be applied if the input is not string (or it's not supported?)

```
>>> j.data.hash.sha1(b'12121')                                                                              
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/abom/projects/jumpscale/js-next/js-ng/jumpscale/data/hash/HashTool.py", line 58, in sha1
    return hashlib.sha1(string.encode(encode)).hexdigest()
AttributeError: 'bytes' object has no attribute 'encode'

'bytes' object has no attribute 'encode'
```

**Comment by abom** [comment]
Please rename `HashTool.py` to `hash.py`

**Comment by omarelsheekh** [comment]
all is done , pls have a look and merge if everything is ok

File: jumpscale/data/hash/hash.py [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+    
+    Returns:
+        [list] -- [the hashes list]
+    """
+    hashes_list = []
+    for d in get_list_files(root_dir):
+        hashes_list.append(hash_file(d, hash_type))
+    return hashes_list
+
+print(md5('hi').hexdigest()==hashlib.md5(b'hi').hexdigest())
```
[the remark this answers is not in the document]
Comment (omarelsheekh) [reply jumpscale/data/hash/hash.py]: ya ,sorry i forgot to remove them

File: poetry.lock [hunk 2]
```diff
@@ -597,4 +597,4 @@ traitlets = ["9c4bd2d267b7153df9152698efb1050a5d84982d3384a37b2c1f7723ba3e7835",
 urllib3 = ["b246607a25ac80bedac05c6f282e3cdaf3afb65420fd024ac94435cabe6e18d1", "dbe59173209418ae49d485b87d1681aefa36252ee85884c31346debd19463232"]
 virtualenv = ["50e062b385b6e5fc6e5644ffe14e226550109a4c65e725b47ae7e3b978d320b5", "fb528e20ecfce349de1dc2c8cbb394d1189b3f9507aea7273fbfb1c2167a4a4e"]
 wcwidth = ["3df37372226d6e63e1b1e1eda15c594bca98a22d33a23832a90998faa96bc65e", "f4ebe71925af7b40a864553f761ed559b43544f8f71746c2d756c7fe788ade7c"]
-zipp = ["4970c3758f4e89a7857a973b1e2a5d75bcdc47794442f2e2dd4fe8e0466e809a", "8a5712cfd3bb4248015eb3b0b3c54a5f6ee3f2425963ef2a0125b8bc40aafaec"]
+zipp = ["4970c3758f4e89a7857a973b1e2a5d75bcdc47794442f2e2dd4fe8e0466e809a", "8a5712cfd3bb4248015eb3b0b3c54a5f6ee3f2425963ef2a0125b8bc40aafaec"]
```
[the remark this answers is not in the document]
Comment (omarelsheekh) [reply poetry.lock]: the two lines are identical , i can't see any changes

File: tests/data/hash/test_hashtool.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
+import pytest
+from jumpscale.god import j
+import hashlib
+import os
+
+
+@pytest.fixture
+def make_hash():
+    h = j.data.hash
```
[the remark this answers is not in the document]
Comment (abom) [reply tests/data/hash/test_hashtool.py]: ?

File: poetry.lock (hunk 2 above)
[the remark this answers is not in the document]
Comment (abom) [reply poetry.lock]: No, there's a change, and git sees it, you removed the new line at the end of the file

See: https://github.com/js-next/js-ng/pull/10/files#diff-41fe8bebc1a2a52eb5321b759e40b3a8L600

**Pull request merged**